### PR TITLE
Terraform backend init.

### PR DESCRIPTION
**Pull request by dancortes94**
# Terraform Backend Init Script.

#### What does this PR do?

- Adds initialization of the backend from the `awsconfig.py`.
- Moves dependencies into a virtualenv and creates `requirements.txt` file.
- Adds README.md about how to use the script.
- Updated outdated `opt-parse`.
- Updated `.gitignore`.

#### Where should the reviewers start?

- python/awsconfig.py
- python/requirements.py
- python/README.md

#### How should this be manually tested?

- Clone the repository to your local environment.
- Checkout the branch `git checkout feat/terraform-init`
- Change directory to the python directory `cd python`
- Install virtualenv using Python3 `pip3 install virtualenv`
- Create the virtual environment in your python directory `python3 -m venv .`
- Activate your environment `source bin/activate`
- Install dependencies `pip3 install -r requirements.txt`
-  Run the command to update the credentials `python3 awsconfig.py -k <aws_access_key> -s "<aws_secret_access_key>" -p "<profile-name>"`
- Run the command to initialize a lesson backend `python3 awsconfig.py -k <aws_access_key> -s "<aws_secret_access_key>" -p "<profile-name> -i -l <lesson-folder> -e <environment>`

#### Any background context you want to provide?

This script tries to simplify the first steps of the people taking the Terraform Academy. This should be easy to use and help automate the terraform backend init independently of the lesson and the environment required.

#### Checklist

- [X] I added the necessary documentation, if appropriate.
- [X] I reviewed existing Pull Requests before submitting mine.
- [X] It sparks joy.

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Configuration Script
+This script helps you configure your AWS credentials into the requiered files and initialize terraform backends.
+
+To run this script you need to configure first a virtual environment:
+
+```
+cd terraform-academy/python
+pip3 install virtualenv
```
Comment (luisc009) [inline python/README.md:8]: https://twitter.com/alicegoldfuss/status/1116150186827337729 :trollface:

**Comment by chitopunk** [comment]
quick question on the initialization
```
Run the command to initialize a lesson backend python3 awsconfig.py -k <aws_access_key> -s "<aws_secret_access_key>" -p "<profile-name> -i -l <lesson-folder> -e <environment>

```

Do we need to add the keys everytime? and the params for environments should be on lesson03 only right?

File: python/README.md (hunk 1 above)
Comment (dancortes94) [reply python/README.md:8]: Spent quite some time fighting with the virtualenvs and pylint... not proud of it hahaha

**Comment by dancortes94** [comment]
@chitopunk We need the keys to find the user that those keys belong to, also, @charlycastro told me he needed multiple environments, hence the env flag.

**Comment by chitopunk** [comment]
> @chitopunk We need the keys to find the user that those keys belong to, also, @charlycastro told me he needed multiple environments, hence the env flag.

@dancortes94  I believe if we use a .env or something would it easier to use, 

like.. 
```
python3 awsconfig.py -l lesson01 -i
python3 awsconfig.py -l lesson02 -i
python3 awsconfig.py -l lesson03 -i
```

What u think?

**Comment by dancortes94** [comment]
@chitopunk Agreed, will work on this and update the PR.

**Comment by chitopunk** [comment]
I get this after filling up the variables.ini file
``` $ python3 awsconfig.py -l lesson-01 -i
usage: awsconfig.py [-h] [-i] [-l lesson]

optional arguments:
  -h, --help            show this help message and exit
  -i, --init            Init terraform backend of a lesson
  -l lesson, --lesson lesson
                        Lesson to initialize backend, accepted values are
                        lesson01, lesson02 or lesson03
-l, --lesson is required```

**Comment by chitopunk** [comment]
disregard my previous message :D should be lesson01 not lesson-01

**Comment by chitopunk** [comment]
I will approve this for now, let's test :D and fix if needed.

**Review by chitopunk — APPROVED** [review]

**Pull request merged**